x/gov/handler.go

```diff
@@ -1,138 +1,35 @@
 package gov
 
 import (
-	"fmt"
-	"strconv"
-
-	metrics "github.com/armon/go-metrics"
-
-	"github.com/cosmos/cosmos-sdk/telemetry"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/x/gov/keeper"
 	"github.com/cosmos/cosmos-sdk/x/gov/types"
 )
 
 // NewHandler creates an sdk.Handler for all the gov type messages
-func NewHandler(keeper keeper.Keeper) sdk.Handler {
+func NewHandler(k keeper.Keeper) sdk.Handler {
 	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
 		ctx = ctx.WithEventManager(sdk.NewEventManager())
 
+		msgServer := keeper.NewMsgServerImpl(k)
+
 		switch msg := msg.(type) {
 		case *types.MsgDeposit:
-			return handleMsgDeposit(ctx, keeper, msg)
+			res, err := msgServer.Deposit(sdk.WrapSDKContext(ctx), msg)
+			return sdk.WrapServiceResult(ctx, res, err)
 
 		case types.MsgSubmitProposalI:
```

Comment:
```suggestion
		case types.MsgSubmitProposaI:
```

we should delete the `MsgSubmitProposalI` interface. It's no longer needed.